app/models/user.rb

```diff
@@ -1,14 +1,14 @@
 class User < ActiveRecord::Base
-  has_secure_password
+  has_secure_password validations: false
 
   has_many :stems, class_name: "Track", foreign_key: "creator_id"
   has_many :remixes, class_name: "Track", foreign_key: "creator_id"
   has_many :circles, foreign_key: "creator_id"
 
   validates_presence_of :user_name
-  validates_presence_of :email
+  validates_presence_of :email, unless: "sc_token.present?"
 
-  validates_length_of :password, minimum: 6, allow_blank: true
+  validates_length_of :password, minimum: 6, allow_blank: true, unless: "sc_token.present?"
```

Comment:
Line is too long. [91/80]<br>Prefer the new style validations `validates :column, length: value` over `validates_length_of`.